cmd/blockchaincmd/join.go

```diff
@@ -325,8 +325,9 @@ func writeAvagoChainConfigFiles(
 
 func checkIsValidating(subnetID ids.ID, nodeID ids.NodeID, pClient platformvm.Client) (bool, error) {
 	// first check if the node is already an accepted validator on the subnet
-	ctx := context.Background()
 	nodeIDs := []ids.NodeID{nodeID}
+	ctx, cancel := sdkutils.GetAPIContext()
```

Comment:
how do we decide if GetAPIContext or GetAPILargeContext

Reply:
large context is used for txs that has been observed to delay on the blockchain request. typically 
a request that takes more time than expected because of data intensive operation.